src/node_http_parser.cc

```diff
@@ -457,6 +457,7 @@ class Parser : public AsyncWrap {
     // Should always be called from the same context.
     CHECK_EQ(env, parser->env());
     parser->Init(type);
+    parser->InitAsyncWrap(env, args.This(), AsyncWrap::PROVIDER_HTTPPARSER);
```

Comment:
Would we have some way to access `parent` at this point, that we could start providing it through the `Init`?

Presently, if you disable _async_wrap_, you don't see the HTTPParser come through, because our parent is null. If we could tie it to the parent server, it would respect the rules a bit better.

*shrug*